analysis/import/SCMLogParser/src/main/kotlin/de/maibornwolff/codecharta/importer/scmlogparser/converter/ProjectConverter.kt

```diff
@@ -42,12 +42,13 @@ class ProjectConverter(private val containsAuthors: Boolean) {
         return edge
     }
 
-    fun convert(versionControlledFiles: List<VersionControlledFile>, metricsFactory: MetricsFactory): Project {
+    fun convert(versionControlledFiles: MutableMap<String, VersionControlledFile>?, metricsFactory: MetricsFactory): Project {
         val projectBuilder = ProjectBuilder()
 
-        versionControlledFiles
-                .filter { vc -> !vc.markedDeleted() }
-                .forEach { vcFile -> addVersionControlledFile(projectBuilder, vcFile) }
+        // changed from master versionControlledFiles.filter -> versionControlledFiles.values.filter
+        //TODO markedDeleted done in previous steps? maybe not needed here
+        versionControlledFiles?.values?.filter { vc -> !vc.markedDeleted() }
+            ?.forEach { vcFile -> addVersionControlledFile(projectBuilder, vcFile) }
```

Comment:
The filtering is not necessary anymore. We mutate the versionControlledFiles directly.

```suggestion
        versionControlledFiles?.values?.forEach { vcFile -> addVersionControlledFile(projectBuilder, vcFile) }
```